### ISSUE TITLE
Anchoring Animated Sprites onto Animated Sprites

### ISSUE THREAD

**Issue by lillian-lemmer** [hypatia-engine/hypatia#79]
Let's say you have a character sprite (we'll call it _parent sprite_). Now let's say you have a detail/child sprite to anchor to the character sprite (we'll call it _child sprite_). If you animate the _child sprite_ so that each frame should line up with another frame, logically (not particularly positionally), it won't work, because there is no synchronization for frames between sprites.

Imagine a _child animation_ of a sword twirling, and a _parent animation_ of a hand twirling a blade. Well, if the animations aren't sync'd it'll look weird when the hand looks like it's cocking the blade at a 45 degree angle, but the _child animation_ is still on the frame which puts it at a 20 degree angle.

This should be an optional and additional feature to Hypatia--the ability to synchronize frames, and perhaps _child walkabouts_ by default.

TL;DR I used to think this was a bug, but I really over-thought this one; but hey, now it's documented for a future feature.

**Comment by lillian-lemmer** [comment]
Jeez, okay so the above is only half-true: here's some more information: this is actually _apparently_ caused by child walkabouts' action/direction being set AFTER we fetched the child walkabout's current animation.

**Comment by lillian-lemmer** [comment]
I believe that the child walkabout animations timedeltas will always be the same to their siblings, descendants AND their parents, because the timedelta used for updating animations is the time _between_ ticks, and the clock is only ticked every iteration of the main loop.

**Comment by lillian-lemmer** [comment]
Currently working on making a slime girl (editing out her bow) and making the bow a child walkabout and assuring it synchronizes and anchor perfectly to provide a proof of concept.

**Comment by lillian-lemmer** [comment]
I'm done with the proof of concept and I kinda wanted to document the fact that all sprites will naturally be synchronized due to the way clock ticks work. Will update soon.

**Comment by lillian-lemmer** [comment]
https://github.com/hypatia-engine/hypatia/commit/306e20ae585e00b5322c961aa4c95882f723ee23
